.github/workflows/pr-size-labeler.yml

```
@@ -0,0 +1,73 @@
+name: PR Size Labeler
+
+on:
+  pull_request_target:
+    branches: main
+
+jobs:
+  authorization-check:
+    permissions: read-all
+    runs-on: ubuntu-latest
+    outputs:
+      approval-env: ${{ steps.collab-check.outputs.result }}
+    steps:
+      - name: Collaborator Check
+        uses: actions/github-script@v8
+        id: collab-check
+        with:
+          result-encoding: string
+          script: |
+            try {
+              const permissionResponse = await github.rest.repos.getCollaboratorPermissionLevel({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                username: context.payload.pull_request.user.login,
+              });
+              const permission = permissionResponse.data.permission;
+              const hasWriteAccess = ['write', 'admin'].includes(permission);
+              if (!hasWriteAccess) {
+                console.log(`User ${context.payload.pull_request.user.login} does not have write access`);
+                return "manual-approval"
+              } else {
+                console.log(`Verified ${context.payload.pull_request.user.login} has write access. Auto approving.`)
+                return "auto-approve"
+              }
+            } catch (error) {
+              console.log(`${context.payload.pull_request.user.login} does not have write access.`)
+              return "manual-approval"
+            }
+
+  size-label:
+    runs-on: ubuntu-latest
+    needs: authorization-check
+    if: needs.authorization-check.outputs.approval-env == 'auto-approve'
+    permissions:
+      contents: read
+      pull-requests: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Calculate PR size and apply label
+        uses: actions/github-script@v8
+        with:
+          script: |
+            const pr = context.payload.pull_request;
+            const totalChanges = pr.additions + pr.deletions;
```

What about accounting for file moves by using something like:

```suggestion
            const totalChanges = Math.max(pr.additions - pr.deletions, pr.additions, pr.deletions);
```

(not sure if Math.max is valid here, might need to test)

---

I feel like this discounts the cognitive burden of deletions/refactors. To start, do you want to just focus on additions?

---

also `pr.additions - pr.deletions` will always be less than or equal to `pr.additions` lol

---

> I feel like this discounts the cognitive burden of deletions/refactors. To start, do you want to just focus on additions?

That's fair - I presume moves aren't considered +100/-100 though?

> also pr.additions - pr.deletions will always be less than or equal to pr.additions lol

LOL too; let's keep as-is